helper/fairshare/jobmanager.go

```diff
@@ -203,57 +285,48 @@ func (j *JobManager) assignWork() {
 				default:
 				}
 
-				job := j.getNextJob()
+				job, queueID := j.getNextJob()
 				if job != nil {
-					j.workerPool.dispatch(job)
+					j.workerPool.dispatch(job,
+						func() {
+							j.incrementWorkerCount(queueID)
+						},
+						func() {
+							j.decrementWorkerCount(queueID)
+						})
 				} else {
 					break
 				}
 			}
 
-			// listen for a wake-up when an emtpy job manager has been given
-			// new work
 			select {
 			case <-j.quit:
 				j.wg.Done()
 				return
 			case <-j.newWork:
-				break
+				// listen for wake-up when an empty job manager has been given work
+			case <-time.After(50 * time.Millisecond):
```

Comment:
happy to discuss this bit

Reply:
one doubt here. if the job manager is periodically waking up to check if there is work available, is the newWork channel still required ?

Reply:
new work will save it from having to wait if the queues are running really low and new work is being added about as fast as it can be consumed. really though, i'd love to get rid of the timer channel to begin with. going to give it a bit more thought, and open to ideas if suggested

Reply:
granted, in that case it probably isn't a huge deal, buuuuutttt....

Reply:
yeah not removing the newWork makes sense, it does shave some ms off a time sensitive process, if i can think of any suggestions to make this more elegant shall let you know! ( i tried thinking about it a little, but all thoughts were grossly complex for the problem at hand )

Reply:
yeah i could probably take some time to re-build the signaling stuff to make it more cohesive, but when i wrote this i didn't see a simple close-to-drop-in solution at the time. i'd really like to get rid of the time bit, unless it makes the code really complex

Reply:
roger. so this is the first solution off the top of my head, nothing fancy so i bet you might've already considered something similar
- Make the newWork channel be a `processableWork` channel, indicating that work is available and can be processed. This would be for both scenarios when nothing is being processed, or work has been lined up and a queue just got unsaturated
- The place where an empty struct is passed in, in `AddJob`, stays the same
- In `getNextQueue`, if all queues are non-empty and saturated, set a flag value protected with a write lock
- In the cleanup function `decrement..` passed to `dispatch`, we can check that flag, if it is set and the queue just got unsaturated, unset the flag and pass in a struct to the `processableWork` channel
I haven't considered feasibilities like `can everything access the things i've set out there`, `does combo-ing newWork into this make sense` and `will this even work`, so it may not be a great idea to rabbithole this, but bleh i had fun thinking this through lol